src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Microsoft.Extensions.Configuration.Binder.SourceGeneration.csproj

```diff
@@ -18,6 +18,10 @@
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="$(MicrosoftCodeAnalysisVersion_4_8)" PrivateAssets="all" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == 'netstandard2.0'">
+    <EmbeddedResource Include="$(CoreLibSharedDir)System\Diagnostics\CodeAnalysis\NullableAttributes.cs" Link="Resources\NullableAttributes.cs" />
+  </ItemGroup>
```

Comment:
I fixed my [original CS0436](https://github.com/dotnet/runtime/pull/101336#issuecomment-2067733988), but got a new one.

If I unconditionally inject `NullableAttributes.cs` it causes a build failure:
```text
artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net462\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\BindingExtensions.g.cs(225,58): error CS0436: The type 'NotNullIfNotNullAttribute' in 'artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net462\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\NullableAttributes.cs' conflicts with the imported type 'NotNullIfNotNullAttribute' in 'Microsoft.Extensions.Configuration.Binder, Version=9.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. Using the type defined in 'artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net462\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\NullableAttributes.cs'. [src\libraries\Microsoft.Extensions.Configuration.Binder\tests\SourceGenerationTests\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests.csproj::TargetFramework=net462]
artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net462\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\BindingExtensions.g.cs(245,58): error CS0436: The type 'NotNullIfNotNullAttribute' in 'artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net462\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\NullableAttributes.cs' conflicts with the imported type 'NotNullIfNotNullAttribute' in 'Microsoft.Extensions.Configuration.Binder, Version=9.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. Using the type defined in 'artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net462\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\NullableAttributes.cs'. [src\libraries\Microsoft.Extensions.Configuration.Binder\tests\SourceGenerationTests\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests.csproj::TargetFramework=net462]
artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net9.0\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\BindingExtensions.g.cs(229,58): error CS0436: The type 'NotNullIfNotNullAttribute' in 'artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net9.0\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\NullableAttributes.cs' conflicts with the imported type 'NotNullIfNotNullAttribute' in 'System.Runtime, Version=9.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Using the type defined in 'artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net9.0\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\NullableAttributes.cs'. [src\libraries\Microsoft.Extensions.Configuration.Binder\tests\SourceGenerationTests\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests.csproj::TargetFramework=net9.0]
artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net9.0\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\BindingExtensions.g.cs(249,58): error CS0436: The type 'NotNullIfNotNullAttribute' in 'artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net9.0\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\NullableAttributes.cs' conflicts with the imported type 'NotNullIfNotNullAttribute' in 'System.Runtime, Version=9.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Using the type defined in 'artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Debug\net9.0\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\NullableAttributes.cs'. [src\libraries\Microsoft.Extensions.Configuration.Binder\tests\SourceGenerationTests\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests.csproj::TargetFramework=net9.0]
```

But without it I get a test failure under `net462`:
```text
    Microsoft.Extensions.SourceGeneration.Configuration.Binder.Tests.ConfigurationBindingGeneratorTests.GetValue_T_Key_DefaultValue [FAIL]
      Generator caused diagnostic in output compilation: 'NotNullIfNotNullAttribute' is inaccessible due to its protection level.
      Stack Trace: ...
    Microsoft.Extensions.SourceGeneration.Configuration.Binder.Tests.ConfigurationBindingGeneratorTests.GetValue_TypeOf_Key_DefaultValue [FAIL]
      Generator caused diagnostic in output compilation: 'NotNullIfNotNullAttribute' is inaccessible due to its protection level.
      Stack Trace: ...
    Microsoft.Extensions.SourceGeneration.Configuration.Binder.Tests.ConfigurationBindingGeneratorTests.GetValue [FAIL]
      Generator caused diagnostic in output compilation: 'NotNullIfNotNullAttribute' is inaccessible due to its protection level.
      Stack Trace: ...
```

Unsure how to proceed.